src/controllers/SliderMenuControllerMobile.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Slider menu controller mobile
+ * @param $
+ * @param selector
+ * @param sliderWidget
+ * @param settings
+ * @param adguardRulesConstructor
+ * @param localization
+ * @param addRule
+ * @returns {{init: init}}
+ * @constructor
+ */
+/* global Ioc, CommonUtils */
+var SliderMenuControllerMobile = function ($, selector, sliderWidget, settings, adguardRulesConstructor, localization, addRule) { // jshint ignore:line
+    var contentDocument = null;
+    var selectedElement = null;
+    var iframeCtrl = Ioc.get('iframeController');
+
+    var nodeParentsCount = 0;
+    var nodeChildsCount = 0;
+    var parents, children;
+    var nodeNumber = 0;
+
+    /*
+     Called from IframeController._showMenuItem to initialize view
+     */
+    var init = function (iframe, options) {
+        selectedElement = options.element;
+        contentDocument = iframe.contentDocument;
+        bindEvents();
+        createSlider();
+        onScopeChange();
+        selector.selectElement(selectedElement);
+
+        children = CommonUtils.getAllChildren(selectedElement);
+        parents = CommonUtils.getParentsLevel(selectedElement);
+
+        parents.splice(0,0,selectedElement);
+
+        nodeParentsCount = parents.length;
+        nodeChildsCount = children.length;
+    };
+
+    var close = function () {
+        iframeCtrl.removeIframe();
+    };
+
+    var plus = function() {
+        (nodeNumber + 1) > nodeParentsCount ? nodeNumber = nodeNumber : nodeNumber++;
+
+        if (nodeNumber >= 0) {
+            if (parents[nodeNumber]) onSliderMove(parents[nodeNumber]);
+        } else {
+            if (children[~nodeNumber]) onSliderMove(children[~nodeNumber]);
+        }
+    };
+
+    var minus = function() {
+        nodeNumber <= -nodeChildsCount ? nodeNumber = nodeNumber : nodeNumber--;
+
+        if (nodeNumber >= 0) {
+            if (parents[nodeNumber]) onSliderMove(parents[nodeNumber]);
+        } else {
+            if (children[~nodeNumber]) onSliderMove(children[~nodeNumber]);
+        }
+    };
+
+    var openSettings = function() {
+        iframeCtrl.showSettingsMenu();
+    };
+
+    var bindEvents = function () {
+        var menuEvents = {
+            '.close': close,
+            '#ExtendedSettingsText': expandAdvanced,
+            '#adv-settings': onScopeChange,
+            '#adg-cancel': iframeCtrl.showSelectorMenu,
+            '#adg-preview': showPreview,
+            '#adg-accept': blockElement,
+            '.adg-slide-btn--plus': plus,
+            '.adg-slide-btn--minus': minus,
+            '.element-rule_more--mobile': openSettings
+        };
+        Object.keys(menuEvents).forEach(function (item) {
+            $(contentDocument.querySelectorAll(item)).on('click', menuEvents[item]);
+        });
+    };
+
+    var blockElement = function () {
+        addRule(getFilterRuleInputText());
+        iframeCtrl.removeIframe();
+        CommonUtils.reloadPageBypassCache();
```

Comment:
Page reloading is unnecessary, let's move it to the default `addRule` implementation.